src/chat/chat_params.rs

```diff
@@ -0,0 +1,289 @@
+use makepad_widgets::*;
+
+use crate::data::store::Store;
+
+live_design! {
+    import makepad_widgets::base::*;
+    import makepad_widgets::theme_desktop_dark::*;
+
+    import crate::shared::styles::*;
+    import crate::shared::widgets::FadeView;
+    import crate::shared::widgets::MoxinButton;
+    import crate::shared::widgets::MoxinSlider;
+    import crate::shared::widgets::MoxinSwitch;
+    import crate::shared::widgets::MoxinTextInput;
+    import makepad_draw::shader::std::*;
+
+
+    // TODO: Use proper icons for right panel.
+    ICON_CLOSE_PANEL = dep("crate://self/resources/icons/open_left_panel.svg")
+    ICON_OPEN_PANEL = dep("crate://self/resources/icons/close_left_panel.svg")
+
+    ChatParamsActions = <View> {
+        height: Fit
+        flow: Right
+
+        <View> {
+            width: Fill
+            height: Fit
+        }
+
+
+        close_panel_button = <MoxinButton> {
+            width: Fit,
+            height: Fit,
+            icon_walk: {width: 20, height: 20},
+            draw_icon: {
+                svg_file: (ICON_CLOSE_PANEL),
+                fn get_color(self) -> vec4 {
+                    return #475467;
+                }
+            }
+        }
+
+        open_panel_button = <MoxinButton> {
+            width: Fit,
+            height: Fit,
+            visible: false,
+            icon_walk: {width: 20, height: 20},
+            draw_icon: {
+                svg_file: (ICON_OPEN_PANEL),
+                fn get_color(self) -> vec4 {
+                    return #475467;
+                }
+            }
+        }
+    }
+
+    ChatParams = {{ChatParams}} {
+        flow: Overlay,
+        width: Fit,
+        height: Fill,
+
+        main_content = <FadeView> {
+            width: 300
+            height: Fill
+            <View> {
+                width: Fill
+                height: Fill
+                padding: {top: 70, left: 25.0, right: 25.0}
+                spacing: 35
+                flow: Down
+                show_bg: true
+                draw_bg: {
+                    color: #F2F4F7
+                }
+
+                <Label> {
+                    draw_text: {
+                        text_style: <BOLD_FONT>{font_size: 12}
+                        color: #000
+                    }
+                    text: "Inference Parameters"
+                }
+
+                <View> {
+                    flow: Down
+                    spacing: 24
+
+                    temperature = <MoxinSlider> {
+                        default: 1.0
+                        text: "Temperature"
+                        min: 0.0
+                        max: 2.0
+                    }
+
+                    top_p = <MoxinSlider> {
+                        text: "Top P"
+                        min: 0.0
+                        max: 1.0
+                    }
+
+                    <View> {
```

Comment:
Idk if there is a better way to have the text on the left and the switch on the right without a view.